src/logcollector/read_multiline.c

```diff
@@ -93,7 +90,11 @@ void *read_multiline(logreader *lf, int *rc, int drop_it) {
             buffer_size++;
         }
 
-        strncpy(buffer + buffer_size, str, OS_MAXSTR - buffer_size - 2);
+        snprintf(buffer + buffer_size, sizeof(buffer) - buffer_size, "%s", str);
+
+        if (OS_MAX_LOG_SIZE - 1 - buffer_size < strlen(str)) {
+            merror("Large message size from file '%s' (length = " FTELL_TT "): '%s'...", lf->file, FTELL_INT64 buffer_size + strlen(str), buffer);
+        }
```

Comment:
What's your opinion about using the snprinf to determinate if there was a truncation?
In fact, if this chunk of message does not fit into the buffer, we can send it so far and, show the error and set _ms variable true, in order to discard the line content until next one